### ISSUE TITLE
Add support for Managed Identity login

### ISSUE THREAD

**Issue by mikkoc** [microsoft/AzureStorageExplorer#2142]
**Is your feature request related to a problem? Please describe.**
We need to access StorageExplorer from inside a VM with assigned Managed Identity. Currently there is no option to login to StorageExplorer with a managed identity.

**Describe the solution you'd like**
Add an option to setup authentication through the Managed Identity.

**Comment by JasonYeMSFT** [comment]
To confirm, are you asking to use enterprise app/service principal to log in to Storage Explorer instead of your user credential?

**Comment by mikkoc** [comment]
No, I would like to use Managed Identity: https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview
To be more specific, in our case, it's a **user-assigned managed identity**

We followed this tutorial: https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/tutorial-linux-vm-access-storage#get-an-access-token-and-use-it-to-call-azure-storage

So currently, to access blobs from the VM we use:

```shell
STORAGE_TOKEN=$(curl 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fstorage.azure.com%2F' -H Metadata:true | jq -r .access_token)

curl https://mystorage.blob.core.windows.net/container/my_file.deb -H "x-ms-version: 2017-11-09" -H "Authorization: Bearer $STORAGE_TOKEN" -o /tmp/my_file.deb
```

**Comment by MRayermannMSFT** [comment]
I'll try to look into this. But what confuses me is, it seems like this feature is intended to help a script or app that you wrote authenticate. Storage Explorer is a GUI tool (aka, something for a human being to use in an interactive way). So in what scenario are you wanting to have this ability? Is it something like: "I have a script which uses managed identity to access storage, but it is having permission issues, so I want to use Storage Explorer with the same identity as the script, so I can debug the issues"?

**Comment by mikkoc** [comment]
Sorry, I should have been more clear, the example I gave is for a different use case.

Managed Identity is not something we came up with: it's an Azure mechanism and is the way forward as it allows to get rid of managing Service Principle credentials, rotate them, etc. It's very similar to AWS IAM instance roles, if you're familiar with that.

All Azure services are slowly adding support for Managed Identities, see for example: https://github.com/Azure/AKS/issues/993

The use case is for a human logged into a desktop environment inside an Azure VM. He/She should be able to use Storage Explorer without having to input credentials, as long as the VM has a Managed Identity associated with it that allows access to Storage.

**Comment by mikkoc** [comment]
Just for my understanding, does Storage Explorer use AzCopy under the hood? 

If so, azcopy already supports login with `--identity`, as per: https://docs.microsoft.com/en-us/azure/storage/common/storage-ref-azcopy-login

So it should be trivial to add this in GUI.

**Comment by MRayermannMSFT** [comment]
@mikkoc were/are you working with VMs that have user or system assigned managed identities?

**Comment by DavidLangworthy** [comment]
@MRayermannMSFT 

Looks like this has been bumping around for a while.  I've run into the need again.  Due to our security requirements users and groups do not have direct access to storage, only VMs with lots of logging and policy attached.  The VMs attach via a umi.  

In the fully general all singing all dancing version a UMI is just like an azure account.  But I don't know an easy way to add it into that flow off the top of my head.  

Plenty to cover all of the actual scenarios I can think of is to add a MI option to the radio control on the Attach Account, Container, Queue, Table flows.  The follow on page would be pretty similar, but needs the OID for the UMI.  Login needs tenant id as well, but I expect this can be figured out from the account name.

**Comment by MatanShabtay** [comment]
What the status of this ?

**Comment by kgaska** [comment]
I have the same request - please add possibility to use managed identity as auth mechanism when using Azure Storage Explorer on VM. In our case access to Azure storage account is possible only from a particular VM. VM has managed identity enabled and that service principal has access to containers configured. I see no reason to set up permissions for each user individually when they already were granted access to VM which suppose have an access to Storage Account

**Comment by luisnaranjo733** [comment]
In the age of tightening security stances, this becomes a critical feature for Azure Storage Explorer. Consider the scenario where a user is required to use passwordless Entra ID auth only. They lose the ability to use Azure Storage Explorer from Bastion private VMs entirely when managed identities are the only they can auth to their storage account.

**Comment by husamhilal** [comment]
Any updates on this please?

**Comment by wvannuffele4** [comment]
Is this basic feature being worked on at all? This issue has been open since 2019 for something which seems like it should be trivially easy to implement. Unfortunately the project is also still closed source so there's no opportunity for anyone in the community to contribute either.

We, too, have network-isolated storage accounts where the only access point is a PAM-based VM inside the network segment. At no point do users have the credentials of the account they're authenticated to the VM with. The only sensible way for them to authenticate to storage accounts is via a managed identity linked to that VM.

The alternative options are, frankly, bleak:
- create generic users they can use to auth once on the vm - Yuck! Completely invalidates the concept of PAM, risk of credential leakage, etc.
- tell users of varying technical skill to use a cli or shell to interact with storage accounts
- effectively build our own internal storage explorer. This would be a complete waste of development efforts if every company in need of this functionality develops their own in-house closed-source solution

If clearly there is not enough bandwidth internally to properly support Storage Explorer, can the project not be made open source as it seemingly was intended to be from the start?

**Comment by husamhilal** [comment]
Has this feature been implemented?
